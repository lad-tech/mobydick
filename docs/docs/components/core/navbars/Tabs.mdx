import ReactPlayer from 'react-player';
import Tabs from './assets/Tabs.mov';

<ReactPlayer playing controls url={Tabs} />

Reusable Tabs component

## Example

```tsx
import {useState} from 'react';

import {createStyles, Tabs, useStyles, View} from '@lad-tech/mobydick-core';


const TABS_TEST = [
  {value: 'tab1', label: 'tab1'},
  {value: 'tab2', label: 'tab2'},
  {value: 'tab3', label: 'tab3'},
  {value: 'tab4', label: 'tab4'},
  {value: 'tab5', label: 'tab5'},
  {value: 'tab6', label: 'tab6'},
  {value: 'tab7', label: 'tab7'},
  {value: 'tab8', label: 'tab8'},
  {value: 'tab9', label: 'tab9'},
];

export const TabsWidget = () => {
  const [styles] = useStyles(stylesFn);

  const [activeValue, setActiveValue] = useState<string | number>('tab1');

  return (
    <View style={styles.container}>
      <Tabs
        list={TABS_TEST}
        onPressCommon={item => setActiveValue(item.value)}
        activeValue={activeValue}
      />
    </View>
  );
};

const stylesFn = createStyles(({spaces}) => ({
  container: {
    gap: spaces.Space16,
    alignItems: 'center',
  },
}));

```
## `Props`

### <div class="label required basic">Required</div>**`list`**

| Type                                                                                          |
|:----------------------------------------------------------------------------------------------|
| [tab[]](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/navbars/types.ts#L7) |

Data for tabs

### <div class="label required basic">Required</div>**`activeValue`**

| TYPE             |
|:-----------------|
| string \| number |

The value that will be selected as active tab

### `containerStyle`

| TYPE                                                             |
|:-----------------------------------------------------------------|
| [ViewStyle](https://reactnative.dev/docs/view-style-props#props) |

Custom styles for tabs container

### `contentContainerStyle`

| TYPE                                                             |
|:-----------------------------------------------------------------|
| [ViewStyle](https://reactnative.dev/docs/view-style-props#props) |

Custom styles for tabs content container

### `fontTab`

| Type                                                                                                       | DEFAULT               |
|:-----------------------------------------------------------------------------------------------------------|:----------------------|
| [TypographyProp](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/typography/types.ts#L49) | 'Regular-Tertiary-XS' |

Font for tab

### `fontActiveTab`

| Type                                                                                                       | DEFAULT            |
|:-----------------------------------------------------------------------------------------------------------|:-------------------|
| [TypographyProp](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/typography/types.ts#L49) | 'Regular-White-XS' |

Font for active tab

### `backgroundColorTab`

| Type   | DEFAULT           |
|:-------|:------------------|
| string | colors.BgTertiary |

Background color for tab

### `backgroundColorActiveTab`

| Type   | DEFAULT            |
|:-------|:-------------------|
| string | colors.ElementAccent |

Background color for active tab

### `onPressCommon`

| TYPE                                                                                                         |
|:-------------------------------------------------------------------------------------------------------------|
| (item: [ITab](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/navbars/types.ts#L7)) => void |

OnPress for all tabs

### `styleTab`

| TYPE                                                             |
|:-----------------------------------------------------------------|
| [ViewStyle](https://reactnative.dev/docs/view-style-props#props) |

Custom styles for tab

### `styleActiveTab`

| TYPE                                                             |
|:-----------------------------------------------------------------|
| [ViewStyle](https://reactnative.dev/docs/view-style-props#props) |

Custom styles for active tab
