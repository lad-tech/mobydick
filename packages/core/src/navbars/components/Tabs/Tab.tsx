import {useCallback} from 'react';

import {ITabProps} from '../../types';
import useStyles from '../../../styles/hooks/useStyles';
import TouchableOpacity from '../../../basic/components/TouchableOpacity/TouchableOpacity';
import {TypographyLegacy} from '../../../typography/components/TypographyLegacy/TypographyLegacy';
import {LABELS} from '../../../other';

import stylesCreate from './stylesCreate';

const Tab = (props: ITabProps): JSX.Element => {
  const [styles, theme] = useStyles(stylesCreate);
  const {
    active,
    item,
    fontTab,
    fontActiveTab,
    backgroundColorTab,
    backgroundColorActiveTab,
    styleTab,
    styleActiveTab,
    onPressCommon,
  } = props;

  const backgroundColorActive =
    backgroundColorActiveTab || theme.colors.ElementBase;
  const backgroundColor = backgroundColorTab || theme.colors.BgTertiary;
  const font = fontTab || 'Regular-Tertiary-XS';
  const fontActive = fontActiveTab || 'Regular-White-XS';

  const selectPressable = useCallback(() => {
    if (item.onPress) {
      item.onPress();
    } else if (onPressCommon) {
      onPressCommon(item);
    }
  }, [item.onPress, onPressCommon]);

  return (
    <TouchableOpacity
      onPress={selectPressable}
      accessibilityLabel={LABELS.tab}
      style={[
        styles.tab,
        active ? styleActiveTab : styleTab,
        {
          backgroundColor: active ? backgroundColorActive : backgroundColor,
        },
      ]}>
      {item.leftIcon ? item.leftIcon : null}
      <TypographyLegacy font={active ? fontActive : font}>
        {item.label}
      </TypographyLegacy>
      {item.rightIcon ? item.rightIcon : null}
    </TouchableOpacity>
  );
};

export default Tab;
