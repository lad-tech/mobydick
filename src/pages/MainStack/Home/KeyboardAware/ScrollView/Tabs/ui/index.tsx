import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';

import {KeyboardAwareScrollViewWithTabsScreen} from './KeyboardAwareScrollViewWithTabsScreen';
import {KeyboardAwareScrollViewWithBottomAndTabsScreen} from './KeyboardAwareScrollViewWithBottomAndTabsScreen';

import IRootStackParamList from '@/shared/lib/constants/rootStackParamList';
import {SCREENS} from '@/shared/lib/constants/screens';

const Tab = createBottomTabNavigator<IRootStackParamList>();

export const KeyboardAwareScrollViewTabsScreen = () => {
  return (
    <Tab.Navigator screenOptions={{tabBarHideOnKeyboard: true}}>
      <Tab.Screen
        name={SCREENS.KeyboardAwareScrollViewWithTabs}
        component={KeyboardAwareScrollViewWithTabsScreen}
      />
      <Tab.Screen
        name={SCREENS.KeyboardAwareScrollViewWithBottomAndTabs}
        component={KeyboardAwareScrollViewWithBottomAndTabsScreen}
      />
    </Tab.Navigator>
  );
};
