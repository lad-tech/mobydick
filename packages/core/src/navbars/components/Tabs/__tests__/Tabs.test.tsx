import {render} from '@testing-library/react-native';

import Tabs from '../Tabs';

describe('Tabs', () => {
  const list = [{label: '1', value: 1}];
  const onPress = () => null;

  test('render correctly', () => {
    const {toJSON} = render(<Tabs list={list} activeValue={'1'} />);
    expect(toJSON()).toMatchSnapshot();
  });
  test('render correctly with props', () => {
    const {toJSON} = render(
      <Tabs
        list={list}
        activeValue={'1'}
        onPressCommon={onPress}
        backgroundColorTab={'#000'}
        backgroundColorActiveTab={'#ff0000'}
        contentContainerStyle={{flex: 1}}
        containerStyle={{flex: 2}}
        fontTab={'Regular-Error-L'}
        fontActiveTab={'Regular-White-L'}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
