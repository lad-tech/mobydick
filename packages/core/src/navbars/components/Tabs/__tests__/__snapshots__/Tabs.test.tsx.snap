// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs render correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "alignItems": "center",
          "paddingHorizontal": 20,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
    data={
      [
        {
          "label": "1",
          "value": 1,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={[]}
    style={
      [
        {
          "maxWidth": "100%",
        },
        undefined,
      ]
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityLabel="tab"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#D0D6E0",
              "borderRadius": 8,
              "marginRight": 8,
              "opacity": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 6,
            }
          }
        >
          <Text
            allowFontScaling={true}
            style={
              [
                {
                  "color": "#5E6678",
                  "fontFamily": "Inter-Regular",
                  "fontSize": 12,
                  "lineHeight": 16,
                  "minHeight": 16,
                },
                undefined,
              ]
            }
          >
            1
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Tabs render correctly with props 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "alignItems": "center",
          "paddingHorizontal": 20,
          "paddingVertical": 8,
        },
        {
          "flex": 1,
        },
      ]
    }
    data={
      [
        {
          "label": "1",
          "value": 1,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={[]}
    style={
      [
        {
          "maxWidth": "100%",
        },
        {
          "flex": 2,
        },
      ]
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityLabel="tab"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#000",
              "borderRadius": 8,
              "marginRight": 8,
              "opacity": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 6,
            }
          }
        >
          <Text
            allowFontScaling={true}
            style={
              [
                {
                  "color": "#F54C3D",
                  "fontFamily": "Inter-Regular",
                  "fontSize": 28,
                  "lineHeight": 32,
                  "minHeight": 32,
                },
                undefined,
              ]
            }
          >
            1
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
